controllers/enqueue_instance.go

```diff
@@ -0,0 +1,57 @@
+package controllers
+
+import (
+	taskclusterv1beta1 "github.com/wellplayedgames/taskcluster-operator/api/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type enqueueRequestForInstance struct{}
+
+// Create implements EventHandler
+func (e *enqueueRequestForInstance) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	for _, req := range e.getInstanceReconcileRequest(evt.Meta) {
+		q.Add(req)
+	}
+}
+
+// Update implements EventHandler
+func (e *enqueueRequestForInstance) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	for _, req := range e.getInstanceReconcileRequest(evt.MetaOld) {
+		q.Add(req)
+	}
+	for _, req := range e.getInstanceReconcileRequest(evt.MetaNew) {
+		q.Add(req)
+	}
+}
+
+// Delete implements EventHandler
+func (e *enqueueRequestForInstance) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	for _, req := range e.getInstanceReconcileRequest(evt.Meta) {
+		q.Add(req)
+	}
+}
+
+// Generic implements EventHandler
+func (e *enqueueRequestForInstance) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	for _, req := range e.getInstanceReconcileRequest(evt.Meta) {
```

Comment:
same as previous PR it would be nice if this is a utility around the other utility rather than for-looping every result